### PR TITLE
Use jenkins pom + modernise (spotbugs) and add CI

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/lib-access-modifier/issues/31

I've tested it with a Jenkins plugin and made sure that restrictions still work and so does the skip property

### CODE REVIEW

File: access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/ProtectedExternally.java [hunk 1]
```diff
@@ -47,6 +49,9 @@
 public class ProtectedExternally extends None {
 
     @Override
+    @SuppressFBWarnings(
+            value={"RCN_REDUNDANT_NULLCHECK_OF_NULL_VALUE","RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", "NP_LOAD_OF_KNOWN_NULL_VALUE"},
+            justification="something in this class confuses spotbugs, I can't see anything redundant and the error line is the catch block")
```
Comment (oleg-nenashev) [inline access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/ProtectedExternally.java:54]: Me neither. It is a historical code anyway, can be revisited later

**Review by oleg-nenashev — APPROVED** [review]
LGTM, thanks for the migration legwork!

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// While this isn't a plugin, it is much simpler to reuse the pipeline code for CI
+// allowing easy windows / linux testing and producing incrementals
+// the only feature that buildPlugin has that relates to plugins is allowing you to test against multiple jenkins versions
+buildPlugin(configurations: [
+        [ platform: "linux", jdk: "8", jenkins: null ],
+        [ platform: "linux", jdk: "11", jenkins: null, javaLevel: "8" ],
```
Comment (jglick) [inline Jenkinsfile:6]: Do not use `javaLevel`.

File: access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/ProtectedExternally.java (hunk 1 above)
Comment (jglick) [reply access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/ProtectedExternally.java:54]: There is a known bug in SpotBugs that was resolved sort of recently involving `try`-with-resources and null checks.

File: pom.xml [hunk 3]
```diff
@@ -33,9 +36,9 @@
   </distributionManagement>
 
   <scm>
-    <connection>scm:git:git@github.com/kohsuke/${project.artifactId}.git</connection>
-    <developerConnection>scm:git:ssh://git@github.com/kohsuke/${project.artifactId}.git</developerConnection>
-    <url>http://${project.artifactId}.kohsuke.org/</url>
+    <connection>scm:git:git@github.com/jenkins/lib-${project.artifactId}.git</connection>
+    <developerConnection>scm:git:ssh://git@github.com/jenkinsci/lib-${project.artifactId}.git</developerConnection>
+    <url>https://${project.artifactId}.kohsuke.org/</url>
```
Comment (jglick) [inline pom.xml:41]: This is going to be a problem. We should either delete https://github.com/timja/access-modifier/blob/8f45e2a54be18232f3f9ad0211e6673889b9ce38/pom.xml#L31-L36 and do without Maven site generation, or deploy to this repo and hope that URL still works (which I doubt).

Comment (timja) [reply pom.xml:41]: this site doesn't even exist, https://access-modifier.kohsuke.org/

I repointed it to github in the incrementals PR

File: pom.xml [hunk 4]
```diff
@@ -2,12 +2,13 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.kohsuke</groupId>
-    <artifactId>pom</artifactId>
-    <version>21</version>
+    <groupId>org.jenkins-ci</groupId>
+    <artifactId>jenkins</artifactId>
```
Comment (jglick) [inline pom.xml:6]: @timja reminder that Artifactory deployment is not going to work until you fix https://github.com/jenkins-infra/repository-permissions-updater/pull/1778#discussion_r543656321

**Pull request merged**